core/commands/storage.go

```diff
@@ -917,3 +991,97 @@ This command updates host information and broadcasts to the BTFS network.`,
 		return nil
 	},
 }
+
+type StatusRes struct {
+	Status   string
+	FileHash string
+	Chunks   map[string]ChunkStatus
+}
+
+type ChunkStatus struct {
+	Price  int64
+	Host   string
+	Status string
+}
+
+var storageUploadStatusCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Check storage upload and payment status (From client's perspective).",
+		ShortDescription: `
+This command print upload and payment status by the time queried.`,
+	},
+	Arguments: []cmds.Argument{
+		cmds.StringArg("session-id", true, false, "ID for the entire storage upload session.").EnableStdin(),
+	},
+	Run: func(req *cmds.Request, res cmds.ResponseEmitter, env cmds.Environment) error {
+		status := &StatusRes{}
+		// check and get session info from sessionMap
+		ssID := req.Arguments[0]
+		ss, err := storage.GlobalSession.GetSession(ssID)
+		if err != nil {
+			return err
+		}
+
+		// check if checking request from host or client
+		cfg, err := cmdenv.GetConfig(env)
+		if err != nil {
+			return err
+		}
+		if !cfg.Experimental.StorageClientEnabled {
```

Comment:
Should be disabled if both client and host are disabled right? because you can have status for both client and host?